### PR TITLE
django-scim2: init at 0.17.3

### PR DESCRIPTION
**Pull request by s1341**
Inits the `django-scim2` python module, and its dependency `scim2-filter-parser`

This package provides a SCIMv2 service provider implementation for django.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by StillerHarpo** [comment]
You should squash your commits to just one.

File: pkgs/development/python-modules/django-scim2/default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    owner = "15five";
+    repo = pname;
+    rev = "refs/tags/${version}";
+    sha256 = "sha256-5zdGPpjooiFoj+2OoglXhhKsPFB/KOHvrZWZd+1nZqU=";
+  };
+
+  buildInputs = [ sssd ];
+  propagatedBuildInputs = [
+    django
+    mock
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/django-scim2/default.nix:38]: Why do we mock all the time?

File: pkgs/development/python-modules/django-scim2/default.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    gssapi
+    ldap
+    sssd
+  ];
+
+  nativeBuildInputs = [ pythonRelaxDepsHook ];
+  pythonRelaxDeps = [
+    "django"
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/django-scim2/default.nix:50]: What is with tests and pythonImportsCheck?

File: pkgs/development/python-modules/scim2-filter-parser/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-cXHozE4RCSi/24Pl8KGXgdLYsh8x5ZBdkiQ8UGfyz1k=";
+  };
+
+  buildInputs = [ sly ];
+  pythonImportsCheck = [ "scim2_filter_parser" ];
+  propagatedBuildInputs = [ sly ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/scim2-filter-parser/default.nix:15]: What is with tests?

Comment (s1341) [reply pkgs/development/python-modules/scim2-filter-parser/default.nix:15]: Added.

File: pkgs/development/python-modules/django-scim2/default.nix (hunk 2 above)
Comment (s1341) [reply pkgs/development/python-modules/django-scim2/default.nix:50]: Added.

File: pkgs/development/python-modules/django-scim2/default.nix (hunk 1 above)
Comment (s1341) [reply pkgs/development/python-modules/django-scim2/default.nix:38]: Removed.

**Comment by s1341** [comment]
Applied fixes and squashed.

File: pkgs/development/python-modules/django-scim2/default.nix [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  pythonImportsCheck = [ "django_scim" ];
+
+  checkInputs = [  mock tox ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/django-scim2/default.nix:48]:
```suggestion
  nativeBuildInputs = [
    pythonRelaxDepsHook
  ];
  
  pythonRelaxDeps = [
    "django"
  ];

  pythonImportsCheck = [
    "django_scim"
  ];

  checkInputs = [
    mock
    unittestCheckHook
  ];
```

- Move `nativeBuildInputs` above `buildInputs`
- Expand lists to multiple lines
- Use the `unittestCheckHook` to run the tests explicitly
- Passing `tox` should not be necessary

**Review by mweinelt — CHANGES_REQUESTED** [review]
Please one commit per package touched. Also note that the commit message should use the full attribute, e.g. `python3Packages.django-scim2`, not just `django-scim2`.

File: pkgs/development/python-modules/django-scim2/default.nix (hunk 4 above)
Comment (s1341) [reply pkgs/development/python-modules/django-scim2/default.nix:48]: `unittestCheckHook` does not work, the tests all fail. `tox` did work. What should I do?

File: pkgs/development/python-modules/django-scim2/default.nix [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+, django
+, dateutil
+, scim2-filter-parser
+, gssapi
+, ldap
+, sssd
+
+# tests
+, mock
+, pytest-django
```
Comment (mweinelt) [inline pkgs/development/python-modules/django-scim2/default.nix:17]: unused

Comment (s1341) [reply pkgs/development/python-modules/django-scim2/default.nix:17]: yup. sorry. fixing.

File: pkgs/development/python-modules/scim2-filter-parser/default.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    sly
+  ];
+
+  pythonImportsCheck = [
+    "scim2_filter_parser"
+  ];
+
+  checkInputs = [
+    unittestCheckHook
+  ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/scim2-filter-parser/default.nix:23]: Are the tests even in the pypi package you're pulling?

```
scim2-filter-parser> unittestCheckPhase
scim2-filter-parser> Executing unittestCheckPhase
scim2-filter-parser> 
scim2-filter-parser> ----------------------------------------------------------------------
scim2-filter-parser> Ran 0 tests in 0.000s
scim2-filter-parser> 
scim2-filter-parser> OK
scim2-filter-parser> Finished executing unittestCheckPhase

Comment (s1341) [reply pkgs/development/python-modules/scim2-filter-parser/default.nix:23]: I believe that there are tests... When I put `tox` in the checkInputs, it ran tests.

Comment (mweinelt) [reply pkgs/development/python-modules/scim2-filter-parser/default.nix:23]: I don't see any tests in the sdist:

```
❯ atool -x scim2-filter-parser-0.4.0.tar.gz
scim2-filter-parser-0.4.0/LICENSE.txt
scim2-filter-parser-0.4.0/README.rst
scim2-filter-parser-0.4.0/pyproject.toml
scim2-filter-parser-0.4.0/src/scim2_filter_parser/__init__.py
scim2-filter-parser-0.4.0/src/scim2_filter_parser/ast.py
scim2-filter-parser-0.4.0/src/scim2_filter_parser/attr_paths.py
scim2-filter-parser-0.4.0/src/scim2_filter_parser/lexer.py
scim2-filter-parser-0.4.0/src/scim2_filter_parser/parser.py
scim2-filter-parser-0.4.0/src/scim2_filter_parser/queries/__init__.py
scim2-filter-parser-0.4.0/src/scim2_filter_parser/queries/sql.py
scim2-filter-parser-0.4.0/src/scim2_filter_parser/transpilers/__init__.py
scim2-filter-parser-0.4.0/src/scim2_filter_parser/transpilers/django_q_object.py
scim2-filter-parser-0.4.0/src/scim2_filter_parser/transpilers/sql.py
scim2-filter-parser-0.4.0/setup.py
scim2-filter-parser-0.4.0/PKG-INFO
scim2-filter-parser-0.4.0.tar.gz: extracted to `scim2-filter-parser-0.4.0'

Comment (s1341) [reply pkgs/development/python-modules/scim2-filter-parser/default.nix:23]: ok. So should I get rid of the checkInputs entirely?

Comment (mweinelt) [reply pkgs/development/python-modules/scim2-filter-parser/default.nix:23]: No, instead fetch from a source that has the tests, if there are any.

Comment (s1341) [reply pkgs/development/python-modules/scim2-filter-parser/default.nix:30]: ok. Switched to github tag source

**Comment by s1341** [comment]
Thanks for the merge!

**Pull request merged**